src/components/Particle.astro

```diff
@@ -0,0 +1,44 @@
+---
+import Particles from "astro-particles"
+import type { ISourceOptions } from "tsparticles-engine";
+
+const options: ISourceOptions = {
+    background: {
+        color: "#000"
+    },
+    fullScreen: {
+        zIndex: -1
+    },
+    particles: {
+        number: {
+            value: 100
+        },
+        move: {
+            enable: true
+        }
+    }
+};
+---
+
+<script>
+    import type { Container, Engine } from "tsparticles-engine";
+    import { loadFull } from "tsparticles";
+  //  import { tsParticles } from "tsparticles-engine";
+
+    declare global {
+        interface Window {
+            particlesInit: (engine: Engine) => Promise<void>;
+            particlesLoaded: (container: Container) => void;
+        }
+    }
```

Comment:
Does not seem necessary to create a Window Object, you can just use the `window`